### PR TITLE
WIP: RF: Implement new API (publish)

### PR DESCRIPTION
**Pull request by bpoldrack**

### CODE REVIEW

File: datalad/distribution/publish.py [hunk 1]
```diff
@@ -104,17 +100,24 @@ class Publish(Interface):
...
-        annex_copy_opts=Parameter(
-            args=("--annex-copy-opts",),
-            metavar='OPT_STR',  # better name?
-            doc="options passed to 'annex copy'",
-            constraints=EnsureStr() | EnsureNone(),))
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+        git_opts=git_opts,
+        annex_opts=annex_opts,
+        annex_copy_opts=annex_copy_opts
```
Comment (yarikoptic) [inline datalad/distribution/publish.py]: we would need something like "--skip-failing".  or "--failing=(fail,skip)" (may be also "retry" but not sure).
The thing is that e.g. for openfmri I have some subdatasets which I did initialize, and they are part of the superdataset but not really ready for publication. so their pushing might fail.  Sure I could cleanse it all up, but I think allowing for skipping failing might be a good usability feature to have

**Comment by yarikoptic** [comment]
just  a note that whenever merging master and RFing please do not kill introduced features (such as in #716)

File: datalad/distribution/publish.py (hunk 1 above)
Comment (mih) [inline datalad/distribution/publish.py]: See #819 for such an option.

**Comment by mih** [comment]
Superseded by #820

**Pull request closed without merging**